src/lavinmq/schema.cr

```diff
@@ -290,10 +290,15 @@ module LavinMQ
       buf = uninitialized UInt8[4]
       file.read_at(0, buf.to_slice)
       version = IO::ByteFormat::SystemEndian.decode(Int32, buf.to_slice)
+      if version == 0 # if version is 0, read 8 more bytes(ts) and check if that's also 0. If so, the file is empty, set version to default.
+        raise EmptyFile.new if IO::ByteFormat::SystemEndian.decode(Int64, file.to_slice(4, 8)).zero?
```

Comment:
Want to use `MFile#read_at` to not cause a `mmap` of the whole file. 
```suggestion
        buf = uninitialized UInt8[8]
        file.read_at(0, buf.to_slice)
        raise EmptyFile.new if IO::ByteFormat::SystemEndian.decode(Int64, buf.to_slice).zero?
```

Could also check that it is just zeros this way:

```suggestion
        buf = uninitialized UInt8[8]
        file.read_at(0, buf.to_slice)
        raise EmptyFile.new if buf.all?(&.zero?)
```

And then is a new exception required or could `IO::EOFError` be used? It's not technically true, but in practice.

```suggestion
        buf = uninitialized UInt8[8]
        file.read_at(0, buf.to_slice)
        raise IO::EOFError.new if buf.all?(&.zero?)
```

Reply:
> And then is a new exception required or could IO::EOFError be used? It's not technically true, but in practice.

No, we can use EOFError, that feels cleaner 👍